steps.sh

```diff
@@ -0,0 +1,32 @@
+steps="${step_context}"
+job=$1
+repo_url="https://github.com/$2"
+run_url="https://github.com/$2/actions/runs/$3"
+results=""
+if [ -z "$steps" ]
+then
+  exit 0
+else
+  for k in $(jq -c 'keys[]' <<< "$steps"); do
+    outcome="$(jq -c ".${k}.outcome" <<< "$steps")"
+    if [[ $outcome != \"success\" ]]
+    then
+      if [ -z "$results" ]
+      then
+        results="<p><b>There are failures in the Github Actions pipeline for repo"
+        results="${results}<a href='${repo_url}'>${2}</a></b></p>"
+        results="${results}<table><tr><th>Job</th><th>Step</th><th>State</th></tr>"
+      fi
+      pipeline="$(sed -e 's/^"//' -e 's/"$//' <<<"$k")"
+      outcome="$(sed -e 's/^"//' -e 's/"$//' <<<"$outcome")"
+      results="${results}<tr><td><a href='${run_url}'>${job}</td><td>${pipeline}</td><td>${outcome}</td></tr>"
```

Comment:
This whole HTML generation lacks escaping.

Reply:
Correct. I'll do that